### PR TITLE
Fix

### PR DESCRIPTION
**Pull request by MattiasGradinApendo**

### CODE REVIEW

File: src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java [hunk 1]
```diff
@@ -55,6 +56,10 @@ private boolean isNewCase(String caseNumber) {
 	public void updateCase(CaseObject caseObject) {
 		final var errand = caseDataClient.getErrandById(caseObject.getCaseNumber());
 
+		if (EXCEPTION_NOT_VALID_USER.equals(errand.getUpdatedByClient())){
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java:59]: This code piece is very obscure in my eyes. If I would look at this code one month from now, I wouldn't have a clue of it's purpose. Please make some comments why this code piece is here and also the probable cause of whats missing (refer to missing part in wso2 configuration maybe) when this message/exception is returned.

Its also a good praxis to log errors (especially when they are not straight forward) to have a possibility to check logs and get a hint that the problem is occurring...

Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java:59]: Yeah that is valid, will add some comments ,include a better error message and log the error

File: src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java [hunk 2]
```diff
@@ -7,20 +7,23 @@
...
-
 import se.sundsvall.processengine.parkingpermit.api.model.CaseObject;
+import se.sundsvall.processengine.parkingpermit.integration.camunda.FinalDecisionWorker;
 import se.sundsvall.processengine.parkingpermit.integration.casedata.CaseDataClient;
 
 @Service
 public class CaseService {
 	private static final String CONFLICT_MESSAGE = "En process för ärendet är startad sedan tidigare och kan därför inte startas igen.";
 	private static final String EXCEPTION_NOT_VALID_USER = "UNKNOWN";
+	private static final Logger LOGGER = LoggerFactory.getLogger(FinalDecisionWorker.class);
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java:26]: Wrong class used - change FinalDecisionWorker.class to CaseService.class

File: src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java [hunk 3]
```diff
@@ -56,8 +59,12 @@ private boolean isNewCase(String caseNumber) {
 	public void updateCase(CaseObject caseObject) {
 		final var errand = caseDataClient.getErrandById(caseObject.getCaseNumber());
 
+		/* If WSO2 does not set a header for what application has sent the last update to CaseData
+		 then this value wil be UNKNOWN and that is an error, as we need it to avoid an infinite loop between us
+		and CaseData*/
 		if (EXCEPTION_NOT_VALID_USER.equals(errand.getUpdatedByClient())){
```
Comment (josc04) [inline src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java:65]: Add tests to verify that exception is throws when updateByClient is UNKNOWN

Comment (MattiasGradinApendo) [reply src/main/java/se/sundsvall/processengine/parkingpermit/service/CaseService.java:64]: Fixed that :)

File: src/test/java/se/sundsvall/processengine/parkingpermit/service/CaseServiceTest.java [hunk 4]
```diff
@@ -43,6 +44,7 @@ class CaseServiceTest {
 	private static final String INSTANCE_ID = UUID.randomUUID().toString();
 	private static final String CASE_NUMBER = "caseNumber";
 	private static final String CONFLICT_MESSAGE = "En process för ärendet är startad sedan tidigare och kan därför inte startas igen.";
+	private static final String EXCEPTION_NOT_VALID_USER = "UNKNOWN";
```
Comment (josc04) [inline src/test/java/se/sundsvall/processengine/parkingpermit/service/CaseServiceTest.java:47]: Strange name for this variable. Why not just UNKNOWN_USER?

Comment (MattiasGradinApendo) [reply src/test/java/se/sundsvall/processengine/parkingpermit/service/CaseServiceTest.java:47]: True I will rename it

**Review by josc04 — APPROVED** [review]

**Pull request merged**